cli/CommandExecutor.cpp

```diff
@@ -194,11 +198,14 @@ void executeDescribeTable(
 /**
  * @brief A helper function that executes a SQL query to obtain a scalar result.
  */
-inline TypedValue executeQueryForSingleResult(const std::string &query_string,
-                                               StorageManager *storage_manager,
-                                               QueryProcessor *query_processor,
-                                               SqlParserWrapper *parser_wrapper,
-                                               Foreman *foreman) {
+inline TypedValue executeQueryForSingleResult(
+    const tmb::client_id main_thread_client_id,
+    const tmb::client_id foreman_client_id,
```

Comment:
I would suggest to move `client_id` related arguments before `bus`, due to the same `tmb` context.

```
  const std::string &query_string,
  const tmb::client_id main_thread_client_id,
  const tmb::client_id foreman_client_id,
  tmb::MessageBus *bus,
```